golang/cosmos/x/vlocalchain/vlocalchain.go

```diff
@@ -30,24 +37,22 @@ func NewReceiver(keeper keeper.Keeper) portHandler {
 
 func (h portHandler) Receive(cctx context.Context, str string) (ret string, err error) {
 	var msg portMessage
-	err = json.Unmarshal([]byte(str), &msg)
+	err = conv.UnmarshalJSONString(str, &msg)
 	if err != nil {
 		return
 	}
 
 	switch msg.Type {
 	case "VLOCALCHAIN_ALLOCATE_ADDRESS":
 		addr := h.keeper.AllocateAddress(cctx)
-		var bz []byte
-		if bz, err = json.Marshal(addr.String()); err != nil {
+		if ret, err = conv.MarshalToJSONString(addr.String()); err != nil {
 			return
 		}
-		ret = string(bz)
 
 	case "VLOCALCHAIN_QUERY_MANY":
 		// Copy the JSON messages string into a CosmosTx object so we can
 		// deserialize it with just proto3 JSON.
-		cosmosTxBz := []byte(`{"messages":` + string(msg.Messages) + `}`)
+		cosmosTxBz := bytes.Join([][]byte{messagesObjStart, msg.Messages, objEnd}, []byte{})
```

Comment:
This looks like a change for the worse, especially in the context of a PR that otherwise embraces `string` over `[]byte`.